### PR TITLE
RUMM-924 Add Data Scrubbing and GDPR docs

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

🎓 This PR updates docs with recent APIs introduced since `1.4.x`: tracking consent and data scrubbing.

### How?

To align it a bit more with [`dd-sdk-android` docs](https://github.com/DataDog/dd-sdk-android/tree/master/docs):
* I created separate `gdpr.md` doc,
* added Tracking Consent API description in all feature docs (Logging / Tracing / RUM),
* added Data Scrubbing section in RUM docs,
* added picture and link to RUM docs in repository's `README.md`.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Merge to / after `release/1.5.0` is issued

### CODE REVIEW

File: docs/rum_collection.md [hunk 1]
```diff
@@ -265,6 +280,50 @@ Global.rum.addError(message: "error message.")
 
 For more details and available options, refer to the code documentation comments in `DDRUMMonitor` class.
 
+## Data scrubbing
+
+To modify RUM event attributes before it is send to Datadog or to drop particular event entirely, use event mappers API when configuring the SDK:
```
Comment (acgh) [inline docs/rum_collection.md:285]: I would change the grammar slightly for clarity:
```suggestion
To modify the attributes of a RUM event before it is sent to Datadog or to drop an event entirely, use the event mappers API when configuring the SDK:
```

File: docs/rum_collection.md [hunk 2]
```diff
@@ -265,6 +280,50 @@ Global.rum.addError(message: "error message.")
...
+    }
+    .setRUMResourceEventMapper { resourceEvent in
+        return resourceEvent
+    }
+    .setRUMActionEventMapper { actionEvent in
+        return actionEvent
+    }
+    .build()
+```
+Each mapper is a Swift closure with a signature of `(T) -> T?`, where `T` is a concrete RUM event type. This allows changing portions of the event before it gets send, for example to redact sensitive information in RUM Resource's `url` you may implement custom `redacted(_:) -> String` function and use it in `RUMResourceEventMapper`:
```
Comment (acgh) [inline docs/rum_collection.md:303]: Again some small English tweaks:
```suggestion
Each mapper is a Swift closure with a signature of `(T) -> T?`, where `T` is a concrete RUM event type. This allows changing portions of the event before it gets sent, for example to redact sensitive information in RUM Resource's `url` you may implement a custom `redacted(_:) -> String` function and use it in `RUMResourceEventMapper`:
```

File: docs/rum_collection.md [hunk 3]
```diff
@@ -265,6 +280,50 @@ Global.rum.addError(message: "error message.")
...
+```swift
+.setRUMResourceEventMapper { resourceEvent in
+    var resourceEvent = resourceEvent
+    resourceEvent.resource.url = redacted(resourceEvent.resource.url)
+    return resourceEvent
+}
+```
+Returning `nil` from the mapper will drop the event entirely (it won't be send to Datadog).
```
Comment (acgh) [inline docs/rum_collection.md:311]: Ditto, replace `send` with `sent`.

File: docs/rum_collection.md [hunk 4]
```diff
@@ -265,6 +280,50 @@ Global.rum.addError(message: "error message.")
...
+
+Only some properties on each event can be mutated:
```
Comment (acgh) [inline docs/rum_collection.md:313]: I would tweak the wording a little bit:
```suggestion
Depending on a given event's type, only some specific properties can be mutated:
```

**Review by acgh — COMMENTED** [review]
This looks great to me! I have suggested a few grammar/wording edits but they are minor.
I'm going to give the chance to more reviewers to have a look at this PR before approving.

File: docs/rum_collection.md (hunk 3 above)
Comment (ncreated) [reply docs/rum_collection.md:311]:
```suggestion
Returning `nil` from the mapper will drop the event entirely (it won't be sent to Datadog).
```

File: docs/trace_collection.md [hunk 5]
```diff
@@ -74,7 +76,20 @@ Datadog.initialize(
     {{% /tab %}}
     {{< /tabs >}}
 
-     When writing your application, you can enable development logs. All internal messages in the SDK with a priority equal to or higher than the provided level are then logged to console logs.
+    To be compliant with the GDPR regulation, the SDK requires the `trackingConsent` value at initialization.
+    The `trackingConsent` can be one of the following values:
+
+    - `.pending` - the SDK starts collecting and batching the data but does not send it to Datadog. The SDK waits for the new tracking consent value to decide what to do with the batched data.
+    - `.granted` - the SDK starts collecting the data and sends it to Datadog.
+    - `.notGranted` - the SDK does not collect any data: logs, traces and RUM events will not be send to Datadog.
```
Comment (cswatt) [inline docs/trace_collection.md:84]:
```suggestion
    - `.notGranted` - the SDK does not collect any data: logs, traces, and RUM events are not sent to Datadog.
```

File: docs/trace_collection.md [hunk 6]
```diff
@@ -74,7 +76,20 @@ Datadog.initialize(
...
+
+    To change the tracking consent value after the SDK is initialized, use `Datadog.set(trackingConsent:)` API.
```
Comment (cswatt) [inline docs/trace_collection.md:86]:
```suggestion
    To change the tracking consent value after the SDK is initialized, use the `Datadog.set(trackingConsent:)` API call.
```

File: docs/trace_collection.md [hunk 7]
```diff
@@ -74,7 +76,20 @@ Datadog.initialize(
...
+    The SDK will change its behavior according to the new value, e.g. if the current tracking consent is `.pending`:
```
Comment (cswatt) [inline docs/trace_collection.md:87]:
```suggestion
    The SDK changes its behavior according to the new value. For example, if the current tracking consent is `.pending`:
```

**Review by cswatt — CHANGES_REQUESTED** [review]
a few styleguide changes

**Comment by ncreated** [comment]
Thank you @cswatt - very consistent feedback 👌. All is applied now, please have another look 🙏 🙂.

File: docs/log_collection.md [hunk 8]
```diff
@@ -76,6 +78,19 @@ Datadog.initialize(
...
+
+    - `.pending` - the SDK starts collecting and batching the data but does not send it to Datadog. The SDK waits for the new tracking consent value to decide what to do with the batched data.
+    - `.granted` - the SDK starts collecting the data and sends it to Datadog.
+    - `.notGranted` - the SDK does not collect any data: logs, traces, and RUM events are not sent to Datadog.
+
+    To change the tracking consent value after the SDK is initialized, use the `Datadog.set(trackingConsent:)` API call.
+    The SDK changes its behavior according to the new value. For example, if the current tracking consent is `.pending`:
+
+    - if changed to `.granted`, the SDK will send all current and future data to Datadog;
+    - if changed to `.notGranted`, the SDK will wipe all current data and will not collect any future data.
```
[the remark this answers is not in the document]
Comment (ncreated) [reply docs/log_collection.md:92]: This is just an example - all transitions are possible, but if I feel that describing all in each features's "Initialization" section is too much. We did similar in `dd-sdk-android`. I think we can improve this once our online doc extracts sort of "Guides" section, where we can explain common topics in details.

**Review by cswatt — APPROVED** [review]

File: docs/rum_collection.md [hunk 9]
```diff
@@ -265,6 +280,50 @@ Global.rum.addError(message: "error message.")
...
+    }
+    .setRUMResourceEventMapper { resourceEvent in
+        return resourceEvent
+    }
+    .setRUMActionEventMapper { actionEvent in
+        return actionEvent
+    }
+    .build()
+```
+Each mapper is a Swift closure with a signature of `(T) -> T?`, where `T` is a concrete RUM event type. This allows changing portions of the event before it gets sent. For example to redact sensitive information in RUM Resource's `url` you may implement a custom `redacted(_:) -> String` function and use it in `RUMResourceEventMapper`:
```
Comment (acgh) [inline docs/rum_collection.md:303]: After RUMM-1077 and #415, we need to update that part to note that `RUMViewEventMapper` must return a non-optional `T` and `RUMViewEvent` can't be dropped (they will all be sent to Datadog).

**Review by acgh — CHANGES_REQUESTED** [review]
To align with RUMM-1077, we need to add a special case for `RUMViewEventMapper` in the paragraph about mappers, as it will return a non-optional type as opposed to the other kinds of mappers.

File: docs/rum_collection.md (hunk 9 above)
Comment (ncreated) [reply docs/rum_collection.md:303]: Done 👍. I also included a note on the `view.name` mutability coming in #418.

**Pull request merged**